### PR TITLE
Add aymanbagabas planck keyboard map

### PR DESCRIPTION
**Pull request by aymanbagabas**
Add aymanbagabas planck keymap. Based on Planck via with encoder and audio support.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/aymanbagabas/olkb-planck-rev6.1-aymanbagabas.json
Comment (elpekenin) [inline keyboards/planck/keymaps/aymanbagabas/olkb-planck-rev6.1-aymanbagabas.json:1]: Looks like a VIA's JSON, doesnt belong to QMK's repo

Comment (aymanbagabas) [reply keyboards/planck/keymaps/aymanbagabas/olkb-planck-rev6.1-aymanbagabas.json:1]: Removed and added a link to config in the readme.

**Review by waffle87 — COMMENTED** [review]
Aside from Encoder Map, how does this PR differ from the [existing Planck VIA keymap](https://github.com/qmk/qmk_firmware/tree/master/keyboards/planck/keymaps/via)? Audio is already enabled at the keyboard level, and bootmagic is conditionally enabled when VIA is enabled. (also the flag you have for bootmagic is incorrect, will not have enabled bootmagic, and will have caused a compilation error...)

**Comment by aymanbagabas** [comment]
> Aside from Encoder Map, how does this PR differ from the [existing Planck VIA keymap](https://github.com/qmk/qmk_firmware/tree/master/keyboards/planck/keymaps/via)? Audio is already enabled at the keyboard level, and bootmagic is conditionally enabled when VIA is enabled. (also the flag you have for bootmagic is incorrect, will not have enabled bootmagic, and will have caused a compilation error...)

You're absolutely right. I removed those rules and updated the config. Other than the Encoder Map and the Encoder config, nothing differs from the existing Planck VIA keymap.

**Comment by waffle87** [comment]
The preferred approach would be to simply add encoder map functionality to the existing VIA keymap then.

**Comment by aymanbagabas** [comment]
> The preferred approach would be to simply add encoder map functionality to the existing VIA keymap then.

The encoder map can be different on a case-to-case basis, some people have two encoders, and some have more.

File: keyboards/planck/keymaps/aymanbagabas/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+*/
+// #define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+// #define MIDI_TONE_KEYCODE_OCTAVES 2
+
+// Most tactile encoders have detents every 4 stages
+#define ENCODER_DIRECTION_FLIP
+
+#define ENCODER_RESOLUTION 4
```
Comment (drashna) [inline keyboards/planck/keymaps/aymanbagabas/config.h:47]: This is the default 
```suggestion
```

**Comment by zvecr** [comment]
Unfortunately since this PRs creation, QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

In the near future we will be providing a more user focused workflow for keymaps, but till then the suggestion is to keep changes in your own fork.

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.

**Pull request closed without merging**